In [1]:
!pip install ucimlrepo --quiet

In [2]:
import pandas as pd
from ucimlrepo import fetch_ucirepo

https://archive.ics.uci.edu/dataset/45/heart+disease

In [3]:
heart_disease = fetch_ucirepo(id=45)
df_original = heart_disease.data.original
df= df_original.drop('ca',axis =1)
df.head()

age  sex  cp  trestbps  chol  fbs  restecg  thalach  exang  oldpeak  slope  \
0   63    1   1       145   233    1        2      150      0      2.3      3   
1   67    1   4       160   286    0        2      108      1      1.5      2   
2   67    1   4       120   229    0        2      129      1      2.6      2   
3   37    1   3       130   250    0        0      187      0      3.5      3   
4   41    0   2       130   204    0        2      172      0      1.4      1   

   thal  num  
0   6.0    0  
1   3.0    2  
2   7.0    1  
3   3.0    0  
4   3.0    0

<!-- Subapartat -->
### <b style="font-family:'Segoe UI', 'Lucida Grande', Verdana, sans-serif; color:#00000;"> <span style='color:#009FE3;'></span> DESCRIPCIÓ DE LES VARIABLES </b>

<p style="font-family:'Segoe UI', 'Lucida Grande', Verdana, sans-serif; font-size:15px; line-height:1.6; color:#00000;">
El conjunt de dades consta de la següent informació de 303 pacients:
</p>

#### <b style="font-family:'Segoe UI', 'Lucida Grande', Verdana, sans-serif; color:#00000;"> <span style='color:#009FE3;'>✦ </span><code>age</code> : Edat </b>
<p style="font-family:'Segoe UI', 'Lucida Grande', Verdana, sans-serif; font-size:15px; line-height:1.6; color:#00000;">
L'edat del pacient en anys. Representada per un nombre enter.
</p>

#### <b style="font-family:'Segoe UI', 'Lucida Grande', Verdana, sans-serif; color:#00000;"> <span style='color:#009FE3;'>✦ </span><code>sex</code> : Sexe </b>
<p style="font-family:'Segoe UI', 'Lucida Grande', Verdana, sans-serif; font-size:15px; line-height:1.6; color:#00000;">
Sexe del pacient. Representada per un indicador binari, <code>0</code> correspon al sexe femení i <code>1</code> al sexe masculí.
</p>

#### <b style="font-family:'Segoe UI', 'Lucida Grande', Verdana, sans-serif; color:#00000;"> <span style='color:#009FE3;'>✦ </span><code>cp</code> : Categoria del dolor toràcic </b>
<p style="font-family:'Segoe UI', 'Lucida Grande', Verdana, sans-serif; font-size:15px; line-height:1.6; color:#00000;">
Categorització del dolor toràcic. Cada categoria està representada per un valor enter: <code>1</code> angina típica; <code>2</code> angina atípica; <code>3</code> dolor no anginós; <code>4</code> asimptomàtic.
</p>

#### <b style="font-family:'Segoe UI', 'Lucida Grande', Verdana, sans-serif; color:#00000;"> <span style='color:#009FE3;'>✦ </span><code>trestbps</code> : Pressió arterial en repòs </b>
<p style="font-family:'Segoe UI', 'Lucida Grande', Verdana, sans-serif; font-size:15px; line-height:1.6; color:#00000;">
Mesura de la pressió arterial en repòs, amb unitats de mm Hg, del pacient en el moment d'ingrés hospitalari representat per un enter.
</p>

#### <b style="font-family:'Segoe UI', 'Lucida Grande', Verdana, sans-serif; color:#00000;"> <span style='color:#009FE3;'>✦ </span><code>chol</code> : Colesterol sèric </b>
<p style="font-family:'Segoe UI', 'Lucida Grande', Verdana, sans-serif; font-size:15px; line-height:1.6; color:#00000;">
Mesura del colesterol sèric del pacient en mg/dl.
</p>

#### <b style="font-family:'Segoe UI', 'Lucida Grande', Verdana, sans-serif; color:#00000;"> <span style='color:#009FE3;'>✦ </span><code>fbs</code> : Glucosa en dejú </b>
<p style="font-family:'Segoe UI', 'Lucida Grande', Verdana, sans-serif; font-size:15px; line-height:1.6; color:#00000;">
Mesura de la glucosa en sang en dejú. Variable binària: <code>1</code> si és &gt; 120 mg/dl, <code>0</code> en cas contrari.
</p>

#### <b style="font-family:'Segoe UI', 'Lucida Grande', Verdana, sans-serif; color:#00000;"> <span style='color:#009FE3;'>✦ </span><code>restecg</code> : Electrocardiograma en repòs </b>
<p style="font-family:'Segoe UI', 'Lucida Grande', Verdana, sans-serif; font-size:15px; line-height:1.6; color:#00000;">
Resultats de l'ECG en repòs: <code>0</code> normal, <code>1</code> anomalies de l'ona ST-T, <code>2</code> hipertrofia ventricular esquerra segons els criteris d'Estes.
</p>

#### <b style="font-family:'Segoe UI', 'Lucida Grande', Verdana, sans-serif; color:#00000;"> <span style='color:#009FE3;'>✦ </span><code>thalach</code> : Freqüència cardíaca màxima </b>
<p style="font-family:'Segoe UI', 'Lucida Grande', Verdana, sans-serif; font-size:15px; line-height:1.6; color:#00000;">
Freqüència cardíaca màxima assolida pel pacient durant una prova d'esforç, mesurada en bpm.
</p>

#### <b style="font-family:'Segoe UI', 'Lucida Grande', Verdana, sans-serif; color:#00000;"> <span style='color:#009FE3;'>✦ </span><code>exang</code> : Angina induïda per l'exercici </b>
<p style="font-family:'Segoe UI', 'Lucida Grande', Verdana, sans-serif; font-size:15px; line-height:1.6; color:#00000;">
Indica si el pacient presenta angina induïda durant la prova d'esforç: <code>1</code> sí, <code>0</code> no.
</p>

#### <b style="font-family:'Segoe UI', 'Lucida Grande', Verdana, sans-serif; color:#00000;"> <span style='color:#009FE3;'>✦ </span><code>oldpeak</code> : Depressió ST </b>
<p style="font-family:'Segoe UI', 'Lucida Grande', Verdana, sans-serif; font-size:15px; line-height:1.6; color:#00000;">
Mesura la quantitat de mil·lívolts (<i>mV</i>) en què el segment ST de l’electrocardiograma queda deprimit respecte al nivell basal durant la prova d’esforç.
</p>

#### <b style="font-family:'Segoe UI', 'Lucida Grande', Verdana, sans-serif; color:#00000;"> <span style='color:#009FE3;'>✦ </span><code>slope</code> : Pendent del segment ST </b>
<p style="font-family:'Segoe UI', 'Lucida Grande', Verdana, sans-serif; font-size:15px; line-height:1.6; color:#00000;">
Codifica la forma del segment ST de l’electrocardiograma durant la prova d’esforç. S’interpreta de la manera següent: <code>1</code> ascendent (normal); <code>2</code> pla (pot indicar isquèmia); <code>3</code> descendent (més sospitós d’isquèmia).
</p>

#### <b style="font-family:'Segoe UI', 'Lucida Grande', Verdana, sans-serif; color:#00000;"> <span style='color:#009FE3;'>✦ </span><code>thal</code> : Resultats del test nuclear d'estrès </b>
<p style="font-family:'Segoe UI', 'Lucida Grande', Verdana, sans-serif; font-size:15px; line-height:1.6; color:#00000;">
Resultat del test nuclear amb tal·li: <code>3</code> normal, <code>6</code> defecte fix, <code>7</code> defecte reversible.
</p>

#### <b style="font-family:'Segoe UI', 'Lucida Grande', Verdana, sans-serif; color:#00000;"> <span style='color:#009FE3;'>✦ </span><code>num</code> : Diagnòstic de malaltia coronària </b>
<p style="font-family:'Segoe UI', 'Lucida Grande', Verdana, sans-serif; font-size:15px; line-height:1.6; color:#00000;">
Variable objectiu que representa el nombre de vasos coronaris principals amb estenosi significativa (&gt;50%). Pot prendre valors de <code>0</code> a <code>4</code>, on: <code>0</code> és cap vas afectat (absència de malaltia coronària); <code>1</code> a <code>3</code>és el nombre de vasos amb estenosi significativa; <code>4</code> correspon a malaltia del tronc principal esquerre (afectació greu).
</p>

1. Determina el  tipus de les dades: Numèriques, Categòriques, Categòriques Ordinals 

Done en clase

2. Converteix "num" a 0 per sa i 1 per malalt.

In [17]:
df["Outcome"] = (df['num'] > 0).astype("int")

df.sample(20)

age  sex  cp  trestbps  chol  fbs  restecg  thalach  exang  oldpeak  \
233   74    0   2       120   269    0        2      121      1      0.2   
96    59    1   4       110   239    0        2      142      1      1.2   
75    65    0   3       160   360    0        2      151      0      0.8   
124   65    1   1       138   282    1        2      174      0      1.4   
188   54    1   2       192   283    0        2      195      0      0.0   
225   34    0   2       118   210    0        0      192      0      0.7   
197   45    0   4       138   236    0        2      152      1      0.2   
275   64    1   1       170   227    0        2      155      0      0.6   
267   59    1   3       126   218    1        0      134      0      2.2   
241   41    0   2       126   306    0        0      163      0      0.0   
172   59    0   4       174   249    0        0      143      1      0.0   
201   64    0   4       180   325    0        0      154      1      0.0   
57    41    1   4       110   172    0        2      158      0      0.0   
22    58    1   2       120   284    0        2      160      0      1.8   
86    47    1   3       138   257    0        2      156      0      0.0   
227   67    0   3       152   277    0        0      172      0      0.0   
156   51    1   4       140   299    0        0      173      1      1.6   
4     41    0   2       130   204    0        2      172      0      1.4   
296   59    1   4       164   176    1        2       90      0      1.0   
126   56    0   4       200   288    1        2      133      1      4.0   

     slope  thal  num  Outcome  
233      1   3.0    0        0  
96       2   7.0    2        1  
75       1   3.0    0        0  
124      2   3.0    1        1  
188      1   7.0    1        1  
225      1   3.0    0        0  
197      2   3.0    0        0  
275      2   7.0    0        0  
267      2   6.0    2        1  
241      1   3.0    0        0  
172      2   3.0    1        1  
201      1   3.0    0        0  
57       1   7.0    1        1  
22       2   3.0    1        1  
86       1   3.0    0        0  
227      1   3.0    0        0  
156      1   7.0    1        1  
4        1   3.0    0        0  
296      2   6.0    3        1  
126      3   7.0    3        1

3. Mostra el hist o plot bar per cada tipus de dades

In [18]:
col_mun = ["age","trestbs","chol","thalach","oldpeak"]

df["age"].plot.hist()

ImportError: matplotlib is required for plotting when the default backend "matplotlib" is selected.

4. Triar una mètrica rellevant per cada atribut

5. Mostra la mètrica segons l'atribut.

6. Mostra les gràfiques rellevants segons 